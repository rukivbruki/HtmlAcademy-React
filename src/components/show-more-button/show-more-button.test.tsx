import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import ShowMoreButton from "./show-more-button";

const mockStore = configureStore([]);

const films = [
  {
    id: 1,
    name: `name1`,
    img: `img/name1.jpg`,
    genre: `test1`,
    preview: `test1`,
  },
  {
    id: 2,
    name: `name2`,
    img: `img/name2.jpg`,
    genre: `test2`,
    preview: `test2`,
  },
  {
    id: 3,
    name: `name3`,
    img: `img/name3.jpg`,
    genre: `test3`,
    preview: `test4`,
  },
];

const SHOWN_CARDS_NUMBER = 6;

it(`ShowMoreButton renders correctly`, () => {
  const store = mockStore({
    [NameSpace.GENRES]: {
      genres: [`genre1`, `genre2`],
      genreFilter: `All genres`
    },
    [NameSpace.FILMS]: {
      shownCardsNumber: SHOWN_CARDS_NUMBER,
      films
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <ShowMoreButton filteredFilms={films}/>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
