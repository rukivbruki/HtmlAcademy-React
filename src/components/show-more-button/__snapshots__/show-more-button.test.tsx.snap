// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton renders correctly 1`] = `null`;
